hack/windows/Install-OVS.ps1

```diff
@@ -30,64 +20,45 @@
   Specifies the path of a local SSL package to be used for installation.
 #>
 Param(
-    [parameter(Mandatory = $false)] [string] $DownloadDir,
     [parameter(Mandatory = $false)] [string] $DownloadURL,
     [parameter(Mandatory = $false)] [string] $OVSInstallDir = "C:\openvswitch",
-    [parameter(Mandatory = $false)] [bool] $CheckFileHash = $true,
     [parameter(Mandatory = $false)] [string] $LocalFile,
-    [parameter(Mandatory = $false)] [bool] $ImportCertificate = $true,
     [parameter(Mandatory = $false)] [bool] $InstallUserspace = $true,
     [parameter(Mandatory = $false)] [string] $LocalSSLFile
 )
 
 $ErrorActionPreference = "Stop"
-$OVSDownloadURL = "https://downloads.antrea.io/ovs/ovs-3.0.5-antrea.1-win64.zip"
+$DefaultOVSDownloadURL = "https://downloads.antrea.io/ovs/ovs-3.0.5-antrea.1-win64.zip"
 # Use a SHA256 hash to ensure that the downloaded archive is correct.
-$OVSPublishedHash = '813a0c32067f40ce4aca9ceb7cd745a120e26906e9266d13cc8bf75b147bb6a5'
-$WorkDir = [System.IO.Path]::GetDirectoryName($myInvocation.MyCommand.Definition)
-$OVSDownloadDir = $WorkDir
-$PowerShellModuleBase = "C:\Windows\System32\WindowsPowerShell\v1.0\Modules"
-
-if (!$LocalFile) {
-    $OVSZip = "$OVSDownloadDir\ovs-win64.zip"
-} else {
-    $OVSZip = $LocalFile
-    $DownloadDir = Split-Path -Path $LocalFile
-}
-
-if ($DownloadDir -ne "") {
-    $OVSDownloadDir = $DownloadDir
-}
-
-$InstallLog = "$OVSDownloadDir\install_ovs.log"
-
-if ($DownloadURL -ne "") {
-    $OVSDownloadURL = $DownloadURL
-    # For user-provided URLs, do not verify the hash for the archive.
-    $OVSPublishedHash = ""
+$DefaultOVSPublishedHash = '813a0c32067f40ce4aca9ceb7cd745a120e26906e9266d13cc8bf75b147bb6a5'
+# $MininalVCRedistVersion is the minimal version required by the provided Windows OVS binary. If a higher
+# version of VC redistributable file exists on the Windows host, we can skip the installation.
+$MininalVCRedistVersion="14.12.25810"
+
+$invocationName=$($myInvocation.MyCommand.Name)
+# If the script is called via a Windows Job, the $myInvocation.MyCommand.Definition is the script block.
+# Then it may throw exceptions when called by GetDirectoryName. Use host temp directory as "$WorkDir" instead.
+try {
+    $WorkDir = [System.IO.Path]::GetDirectoryName($myInvocation.MyCommand.Definition)
+} catch {
+    Write-Host "Failed to get the script locations, use temp path as instead"
+    $WorkDir = $env:Temp
 }
+$InstallLog = "$WorkDir\install_ovs.log"
+$PowerShellModuleBase = "C:\Windows\System32\WindowsPowerShell\v1.0\Modules"
+$OVSZip=""
+$DefaultOVSRunDir = "C:\openvswitch\var\run\openvswitch"
+# $global:updateDriver is a flag to mark if the OVSext driver needs to update on the target Windows host or not.
```

Comment:
s/a flag to mark if the OVSext driver needs to update on the target Windows host or not/a flag used to indicate whether the OVSext driver needs to be updated on the target Windows host

Reply:
updated.